docs/aws_cli_config.md

```diff
@@ -0,0 +1,159 @@
+
+```
+mfa_serial = arn:aws:iam::123456789:mfa/whoami@ucsc.edu
+```
+
+Once you have added this MFA device and enabled MFA, test it by running a simple
+list-buckets action. You should be propmted for your 2FA code:
+
+```
+$ aws s3 ls
+Enter MFA code for arn:aws:iam::652235167018:mfa/chmreid@ucsc.edu:
+
+2020-01-17 15:03:22 mah-bukkit-1
+2020-01-17 15:03:22 mah-bukkit-2
+2020-01-17 15:03:22 mah-bukkit-3
+...
+```
+
+## Using Assume Role with Multi-Factor Authentication
+
+Once multi-factor authentication is enabled, every command run using the AWS CLI will
+require a 2FA token. To have `aws` remember your identity across multiple commands,
+use the [assume-role](https://github.com/remind101/assume-role) command line utility.
+(See the Readme for installation instructions.)
+
+Once you have installed `assume-role`, you can pass it the name of a profile, and it
+will output temporary credentials:
+
+```
+$ assume-role my-red-profile
+export AWS_ACCESS_KEY_ID="ASIAI....MPLE"
+export AWS_SECRET_ACCESS_KEY="DuH...MPLE"
+export AWS_SESSION_TOKEN="AQ...1BQ=="
+export AWS_SECURITY_TOKEN="AQ...1BQ=="
+export ASSUMED_ROLE="my-red-profile"
+```
+
+To export these variables, pass the output of the `assume-role` command to `eval`:
+
+```
+$ eval $(assume-role my-red-profile)
+```
+
+This can be assigned to an alias, for example, with bash:
+
+```
+alias pgi='eval $(assume-role gi)'
+```
+
+Each time you call `assume-role` you will be asked to enter your 2FA token, and your
+session will last approximately 30 minutes:
+
+```
+$ eval $(assume-role my-red-profile)
+MFA: ******
+```
+
+Now you can run AWS commands without being prompted for a 2FA token each time.
+
+## Using Multiple Profiles
+
+If you are using the AWS CLI with multiple AWS accounts (profiles), extra care is required
+to be able to switch between profiles. Running `aws configure` again will result in new entries
+in the `config` and `credentials` files.
+
+An example multi-profile `~/.aws/configure` file:
+
+```
+[profile my-red-profile]
+role_arn = arn:aws:iam::123456789:role/red-developer
+role_session_name = whoami@ucsc.edu
+source_profile = red-creds
+
+[profile my-green-profile]
+role_arn = arn:aws:iam::123456789:role/green-developer
+role_session_name = whoami@ucsc.edu
+source_profile = green-creds
+```
+
+Now the credentials file will also have two sets of credentials.
+An example multi-profile `~/.aws/credentials` file:
+
+```
+[red-creds]
+aws_access_key_id = AKIAIOSFODNN7EXAMPLE
+aws_secret_access_key = wJalrXUtnFEMI/K7MDENG/bPxRfiCYEXAMPLEKEY 
+[green-creds]
+aws_access_key_id = AKIAZHOHOHOHOEXAMPLE
+aws_secret_access_key = vy7JKuTKXfgrm/N6MQOZY/cGwPFrymEXAMPLEKEY 
+```
+
+To verify that each profile is working, try running AWS commands
+and specifying the profile.
+
+```
+aws s3 ls --profile my-red-profile
+aws s3 ls --profile my-green-profile
+```
+
+**IMPORTANT:** Firefox has an internal bug that causes problems when accessing the
```

Comment:
https://addons.mozilla.org/en-US/firefox/addon/multi-account-containers/ can also be used :)

Reply:
fixed in 7d86b29